Source/HTTP/httpcall.cpp

```diff
@@ -353,6 +363,20 @@ void HC_CALL::PerformSingleRequestComplete(XAsyncBlock* async)
         }
     }
 
+    // Decompress Response Bytes 
+    if (Compression::Available() && call->compressedResponse == true)
+    {
+        http_internal_vector<uint8_t> uncompressedResponseBodyBuffer;
+
+        Compression::DecompressFromGzip(
+            call->responseBodyBytes.data(),
+            call->responseBodyBytes.size(),
+            uncompressedResponseBodyBuffer);
+
+        call->responseBodyBytes.resize(uncompressedResponseBodyBuffer.size());
```

Comment:
I think here you'll need an extra check to see if the client had specified a response body write callback. If they did, you'd call that here.  If not, you copy the decompressed response to call->responseBodyBytes (like you've done) and they'll read it with HCHttpCallResponseGetResponseBodyBytes after the request completes.

Reply:
Is a call to the custom write callback still needed here? Currently its missing....I'm not quite sure how we would call it I similar to the logic in [http_android.cpp](https://github.com/microsoft/libHttpClient/blob/565b98f779b13321372263b6addeab263cd518e0/Source/HTTP/Android/http_android.cpp#L254)?